### PR TITLE
[pickers] Migrate `MomentHijri` to our repository

### PR DESCRIPTION
**Pull request by flaviendelangle**
And here is the last one

### CODE REVIEW

File: packages/x-date-pickers-pro/README.md [hunk 1]
```diff
@@ -42,15 +42,6 @@ This component has the following peer dependencies that you will need to install
 },
 ```
 
-If you need to use `js-joda`, `date-fns-jalali`, `jalaali`, or `hijri` library, you should be able to find the corresponding date-library from [`@date-io`](https://github.com/dmtrKovalenko/date-io#projects).
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/README.md:45]: We need to rework those README files...

File: packages/x-codemod/tsconfig.json [hunk 2]
```diff
@@ -1,8 +1,6 @@
 {
   "extends": "../../tsconfig",
   "compilerOptions": {
-    // @date-io libraries produce duplicate type `DateType`
-    "skipLibCheck": true,
```
Comment (LukasTy) [inline packages/x-codemod/tsconfig.json:5]: 🙈 👍

File: packages/x-date-pickers-pro/README.md (hunk 1 above)
Comment (LukasTy) [reply packages/x-date-pickers-pro/README.md:45]: Oh yeah, they could use some touch-up or even a DevEx cc @samuelsycamore eye, as they are somewhat visible, especially in npm.js and GH. 🤔

File: packages/x-date-pickers/src/AdapterMomentHijri/AdapterMomentHijri.test.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  });
+
+  describe('Adapter localization', () => {
+    it('Formatting', () => {
+      const adapter = new AdapterMomentHijri();
+
+      const expectDate = (format: keyof AdapterFormats, expectedWithFaIR: string) => {
+        const date = adapter.date('2020-01-01T23:44:00.000Z')!;
+
+        expect(adapter.format(date, format)).to.equal(expectedWithFaIR);
```
Comment (LukasTy) [inline packages/x-date-pickers/src/AdapterMomentHijri/AdapterMomentHijri.test.tsx:33]: Isn't it arSA for this adapter? 🤔 
```suggestion
      const expectDate = (format: keyof AdapterFormats, expectedWithArSA: string) => {
        const date = adapter.date('2020-01-01T23:44:00.000Z')!;

        expect(adapter.format(date, format)).to.equal(expectedWithArSA);
```

File: packages/x-date-pickers/src/AdapterMomentHijri/AdapterMomentHijri.ts [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+      current = current.clone().add(1, 'day');
+      count += 1;
+    }
+
+    return nestedWeeks;
+  };
+
+  public getWeekNumber = (date: Moment) => {
+    return date.iWeek();
```
Comment (LukasTy) [inline packages/x-date-pickers/src/AdapterMomentHijri/AdapterMomentHijri.ts:261]:
```suggestion
  public getWeekNumber = (value: Moment) => {
    return value.iWeek();
```

File: packages/x-date-pickers/src/AdapterMomentHijri/AdapterMomentHijri.ts [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+  public startOfMonth = (date: Moment) => {
+    return date.clone().startOf('iMonth');
+  };
+
+  public endOfYear = (date: Moment) => {
+    return date.clone().endOf('iYear');
+  };
+
+  public endOfMonth = (date: Moment) => {
+    return date.clone().endOf('iMonth');
```
Comment (LukasTy) [inline packages/x-date-pickers/src/AdapterMomentHijri/AdapterMomentHijri.ts:186]:
```suggestion
  public startOfYear = (value: Moment) => {
    return value.clone().startOf('iYear');
  };

  public startOfMonth = (value: Moment) => {
    return value.clone().startOf('iMonth');
  };

  public endOfYear = (value: Moment) => {
    return value.clone().endOf('iYear');
  };

  public endOfMonth = (value: Moment) => {
    return value.clone().endOf('iMonth');
```

**Review by LukasTy — APPROVED** [review]
Great work on adapters! 💯 🚀 
Really cleans everything up and finally opens the option for quicker and easier changes (I wanted to add a few more formats a few times already... 🙈).

**Pull request merged**